src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -44,9 +44,13 @@ partial class CreationOptions {
 		public abstract partial class JniValueManager : ISetRuntime, IDisposable {
 
 			public      JniRuntime  Runtime { get; private set; }
+			protected   bool        disposed = false;
```

Comment:
Normally `disposed` is _not_ `protected`, and I don't believe that it needs to be. `MonoRuntimeValueManager` can use `RegisteredInstances` instead.

Reply:
OK, how about replacing it with `public bool Disposed { get; private set; } = false;` property? That shouldn't do any harm.

Reply:
Or maybe protected `IsDisposed` as here https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.channels.message.isdisposed?view=netcore-2.0

As the `Disposed` name is used for events in .NET frameworks.

Reply:
an `IsDisposed` property won't do "harm" per-se, but I don't like it, and it is *generally* not something that is provided.

See for example:

* [`System.IO.Stream`](https://msdn.microsoft.com/en-us/library/system.io.stream(v=vs.110).aspx)
* [`System.Security.SecureString`](https://msdn.microsoft.com/en-us/library/system.security.securestring(v=vs.110).aspx)
* [`System.Threading.Timer`](https://msdn.microsoft.com/en-us/library/system.threading.timer(v=vs.110).aspx)
* etc.

In fact, I can (*easily*!) find only one place where an `(Is)?Disposed`-like property is present:

```bash
$ pwd
…/mono/mcs/class/corlib
$ git grep -l IDisposable System* | xargs grep Disposed
System.IO.IsolatedStorage/IsolatedStorageFile.cs:		internal bool IsDisposed {
System.IO.IsolatedStorage/IsolatedStorageFile.cs:				throw new ObjectDisposedException ("IsolatedStorageFile");
System.Security/SecureString.cs:					throw new ObjectDisposedException ("SecureString");
System.Security/SecureString.cs:				throw new ObjectDisposedException ("SecureString");
System.Security/SecureString.cs:				throw new ObjectDisposedException ("SecureString");
System.Security/SecureString.cs:				throw new ObjectDisposedException ("SecureString");
System.Security/SecureString.cs:				throw new ObjectDisposedException ("SecureString");
System.Security/SecureString.cs:				throw new ObjectDisposedException ("SecureString");
System.Security/SecureString.cs:				throw new ObjectDisposedException ("SecureString");
```

Note that it's _`internal`_, and all the others are not members.

What about `corefx`?

```bash
$ pwd
…/mono/external/corefx
$ git grep -l IDisposable src | xargs grep 'public.*Disposed' | grep -v tests/
src/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs:        public event System.EventHandler Disposed { add { } remove { } }
src/System.ComponentModel.TypeConverter/ref/System.ComponentModel.cs:        public event System.EventHandler Disposed { add { } remove { } }
src/System.Runtime/ref/System.Runtime.cs:    public partial class ObjectDisposedException : System.InvalidOperationException
src/System.Runtime/ref/System.Runtime.cs:        public ObjectDisposedException(string objectName) { }
src/System.Runtime/ref/System.Runtime.cs:        public ObjectDisposedException(string message, System.Exception innerException) { }
src/System.Runtime/ref/System.Runtime.cs:        public ObjectDisposedException(string objectName, string message) { }
```

Two `public` *events*, not properties. Looking for `protected` members is no better.

TL;DR: Providing public/protected APIs to query the "disposed" status of an instance is ***not** common*. Providing such a method is decidedly *uncommon*, and I don't see any need to provide such an API here.

Reply:
It is not uncommon in the .NET framework. See https://docs.microsoft.com/en-us/dotnet/api/?view=netframework-4.7&term=IsDisposed

Some of these are abstract classes implementing `IDisposable` iface, which provide protected `IsDisposed` property. I think it makes sense to provide a single place to check for disposed status in this case, where we provide the dispose infrastructure as part of the abstraction.

Reply:
Nice search URL; I wasn't aware of that.

That said, it finds *14* matches, *most* of which are `System.Windows` (7), `System.Xaml` (2), and `System.Printing` (2), leaving *3* in a nominally "core" namespace (`System.Runtime`, `System.ServiceModel`).

This is not what I'd call "not uncommon."

Alas, [searching for `IDisposable`](https://docs.microsoft.com/en-us/dotnet/api/?view=netframework-4.7&term=IDisposable) isn't entirely useful, as most matches are for *explicitly implemented* interface methods (bleh), so I guess [searching for `Dispose` is the "better" bet](https://docs.microsoft.com/en-us/dotnet/api/?view=netframework-4.7&term=Dispose). After hitting **Load more results** *a lot* -- and still not hitting any "end" -- there are (at least!) *213* matches for `Dispose() Method`.

Compare to *14*.

This is not what I'd call "not uncommon"; it is *very* uncommon.

Reply:
Most of these 213 matches are not abstract classes implementing `IDisposable`. So I would still say it is not uncommon, to clarify, for abstract classes.

It is not that important though, so I will update the PR and move along.